packages/framework/src/hooks/useRegisterBindings.ts

```diff
@@ -23,7 +24,16 @@ export const useRegisterBindings = <T extends Record<string, unknown>>(
 ): RegisterBindingsResult<T> => {
   const { resolvedWidgetId, rootRef } = useWidgetId(id);
   const [widgetState, setWidgetState] = useWidgetState<T>(resolvedWidgetId);
+  const themeContext = useStyleNames(String(get(values, ["styles", "names"])));
 
+  const styles = merge({}, themeContext, values?.styles);
+  merge(
+    values,
+    !isEmpty(styles) && {
+      ...values,
+      styles,
+    },
+  );
```

Comment:
Just merge once? The function is recursive. There's no point shallow copying the values since merge will keep the original value if the key is undefined. https://lodash.com/docs/4.17.15#merge